src/main/scala/viper/gobra/frontend/info/implementation/property/Visibility.scala

```diff
@@ -0,0 +1,205 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyr (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.frontend.info.implementation.property
+
+import viper.gobra.ast.frontend._
+import viper.gobra.frontend.info.base.{SymbolTable => symb}
+import viper.gobra.frontend.info.implementation.TypeInfoImpl
+import viper.gobra.util.Violation
+
+trait Visibility extends BaseProperty { this: TypeInfoImpl =>
+
+  def isPrivate(expr: PExpression): Boolean = isPrivateExp(expr)
+
+  private def isPrivateExpTyp(etyp: PExpressionOrType): Boolean = etyp match {
+    case exp: PExpression => isPrivateExp(exp)
+    case typ: PType => isPrivateTyp(typ)
+    case t => Violation.violation(s"Unexpected Expression or Type in isPrivateExpTyp, got $t") 
+  }
+
+  def isPrivateReg(id: PIdnNode): Boolean = regular(id) match {
+    case g: symb.GlobalVariable => isPrivateString(g.id.name)
+    case _: symb.Variable => false 
+    case c: symb.Constant => c.decl.left.exists(id => isPrivateString(id.name))
+    case f: symb.Function => isPrivateString(f.decl.id.name) || (if (f.isPure) isPrivateVParam(f.args) else false)
```

Comment:
I corrected that. I checked the arguments, because before pure functions and predicates were always imported with the body, therefore the type of arguments cannot be private.